server/deletion.go

```diff
@@ -155,32 +155,44 @@ func purgeDanglingUserPosts(db *sql.DB, userID string) error {
 }
 
 func purgeEmptyChannels(db *sql.DB, pluginClient *pluginapi.Client, socketClient *model.Client4) error {
-	rows, err := db.Query(`
-		SELECT id FROM Channels
-		WHERE NOT EXISTS (
-			SELECT 1
-			FROM ChannelMembers
-			WHERE ChannelMembers.channelid = Channels.id
-		);
-	`)
-	if err != nil {
-		return err
-	}
-
-	for rows.Next() {
-		var id string
-		if err := rows.Scan(&id); err != nil {
+	for {
+		rows, err := db.Query(`
+			SELECT id FROM Channels
+			WHERE NOT EXISTS (
+				SELECT 1
+				FROM ChannelMembers
+				WHERE ChannelMembers.channelid = Channels.id
+			)
+			LIMIT 1000;
+		`)
+		if err != nil {
 			return err
 		}
+		defer rows.Close()
 
-		resp, err := socketClient.PermanentDeleteChannel(context.Background(), id)
-		if err != nil {
-			return err
+		ids := []string{}
+		for rows.Next() {
+			var id string
+			if err := rows.Scan(&id); err != nil {
+				return err
+			}
+			ids = append(ids, id)
 		}
-		if resp.StatusCode != http.StatusOK {
-			return fmt.Errorf("%d status code during attempt to delete channel %s", resp.StatusCode, id)
+
+		if len(ids) == 0 {
+			break
+		}
```

Comment:
**question:** what does this change do? If there are no rows, wouldn't we get the same behavior with the code we had before?

Reply:
I was doing some testing and I'm not sure why but when I was using `if !rows.Next()` it would only loop once. I didn't debug it because it was easier to just switch to this pattern which I used previously. It could have been some data issues with the users I was deleting but I know this works